src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -619,6 +622,7 @@ public synchronized void open() throws SFException, SnowflakeSQLException {
       TelemetryService.disable();
     }
 
+    securityUtil.addBouncyCastleProvider();
```

Comment:
I need to add the provider before key decryption takes place and before any HTTP requests are made, and I initially chose this location intentionally because the motive was to be able to configure this on a connection level. If we take the JVM argument route then this can be moved elsewhere but I'm thinking it's going to be either here to in the SnowflakeDriver class.

Reply:
I discussed this with @sfc-gh-pfus and moved it to the SnowflakeDriver so it's done during initialization. It only needs to happen once.